pkg/operator/resourcesynccontroller/interfaces.go

```diff
@@ -0,0 +1,19 @@
+package resourcesynccontroller
+
+// ResourceLocation describes coordinates for a resource to be synced
+type ResourceLocation struct {
+	Namespace string
+	Name      string
+}
+
+var emptyResourceLocation = ResourceLocation{}
+
+// ResourceSyncer allows changes to syncing rules by this controller
+type ResourceSyncer interface {
+	// SyncConfigMap indicates that a configmap should be copied from the source to the destination.  It will also
+	// mirror a deletion from the source.  If the source, is a zero object the destination will not be synced.
```

Comment:
If the source is a zero object, ...

Same below. Actually, the logic is just looking at the namespace, isn't it?

Reply:
> f the source is a zero object, ...
> 
> Same below. Actually, the logic is just looking at the namespace, isn't it?

fixed